deepmd/pt/utils/stat.py

```
@@ -35,51 +32,179 @@
+        iterator = iter(dataloader)
+        numb_batches = min(nbatches, len(dataloader))
+        for _ in range(numb_batches):
+            try:
+                stat_data = next(iterator)
+            except StopIteration:
+                iterator = iter(dataloader)
+                stat_data = next(iterator)
+            for dd in stat_data:
+                if stat_data[dd] is None:
+                    sys_stat[dd] = None
+                elif isinstance(stat_data[dd], torch.Tensor):
```

Ruff (0.8.2)

Undefined name `torch`

(F821)